### PR TITLE
[CUR-826] Added lot picker to pathways

### PR DESCRIPTION
**Pull request by sonalisin**
## Description

Music year: 1950

- Replaced `examboard` in `subjectPhaseOptions` with `ks4_options` to also include pathway
- Added sorted ks4Options into lot picker
- Updated tests and fixtures to use `ks4Options` instead of `examboard` (only where relates to lot picker, did not do this for visualiser / `curriculumUnits`
- Updated tests to remove `All examboards` in the CurriculumHeader as this option is no longer available
- Styling has not been updated yet as waiting for lot picker changes to be merged 

## How to test

1. Go to https://deploy-preview-2805--oak-web-application.netlify.thenational.academy
2. For running locally, ensure that the constants.ts file has Cycle 2 subjects enabled:
  ```export const ENABLE_CYCLE_2 = true;```
3. Click on lot picker - PE, Computing, Citizenship, RE
4. You should see Core +/ GCSE options
5. RHSE should have no options as this isn't available
6. Computing should have [Examboard] (Computer Science)

## Screenshots

How it should now look:
Screenshot 2024-09-19 at 18 43 45

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/CurriculumComponents/CurriculumHeader/CurriculumHeader.tsx [hunk 1]
```diff
@@ -136,10 +138,18 @@ const CurriculumHeader: FC<CurriculumHeaderPageProps> = ({
             ]}
           />
           <Hr $color={"white"} />
-          <SubjectPhasePicker
-            {...subjectPhaseOptions}
-            currentSelection={currentSelection}
-          />
+          {!cycleTwoEnabled && (
```
Comment (orangemug) [inline src/components/CurriculumComponents/CurriculumHeader/CurriculumHeader.tsx:141]: These two look the same?

File: src/utils/curriculum/constants.ts [hunk 2]
```diff
@@ -1 +1 @@
-export const ENABLE_CYCLE_2 = false;
+export const ENABLE_CYCLE_2 = true;
```
Comment (orangemug) [inline src/utils/curriculum/constants.ts:1]: We don't need to enable this, James is happy to test locally for now.

**Review by rightsaidjames — APPROVED** [review]
- [ ]  When a user chooses ‘Secondary’ for a subject with pathways:
    - [x]  They see this copy: ‘Choose an option for KS4’ - done in [[Lot picker: Phase picker modal](https://www.notion.so/Lot-picker-Phase-picker-modal-46f8254eecea43298774f81e6b881d10?pvs=21)](https://www.notion.so/Lot-picker-Phase-picker-modal-46f8254eecea43298774f81e6b881d10?pvs=21)
    - [x]  They see buttons for ‘Core’ and each of the exam boards for that particular subject.
    - [ ]  If there are no exam boards for that subject at KS4, there is a ‘GCSE’ button instead.
- [x]  When a user has chosen ‘Secondary’ and ‘Core’ from the lot picker:
    - [x]  Core units are shown for Y10 and Y11
    - [x]  GCSE units are not shown.
- [ ]  When a user has chosen ‘Secondary’ and an exam board from the lot picker (assuming that the subject has pathways):
    - [ ]  Core units are not shown for Y10 and Y11
    - [ ]  GCSE units are shown, and are tailored to the chosen exam board (where relevant)
- [ ]  When a user has chosen ‘Secondary’ and ‘Core from the lot picker:
    - [ ]  Core units are not shown for Y10 and Y11
    - [ ]  GCSE units are shown

Outstanding issues:
- Subjects with pathways but no exam board variance do not show a ‘GCSE’ option in the lot picker
- GCSE pathways (e.g. Computer Science) are showing Core units
- RE (which has an empty core programme) is not displaying any KS4 units in the visualiser

A new ticket has been created to resolve this so that this PR can be merged.

**Review by orangemug — APPROVED** [review]

**Comment by orangemug** [comment]
@rightsaidjames I've approved this PR, however can you give it another QA smoke test as it was a nasty merge conflict.

**Pull request merged**